ibm/service/appconfiguration/resource_ibm_app_config_segment.go

```diff
@@ -0,0 +1,346 @@
+					Schema: map[string]*schema.Schema{
+						"attribute_name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Attribute name.",
+						},
+						"operator": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Operator to be used for the evaluation if the entity belongs to the segment.",
+						},
+						"values": {
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "List of values. Entities matching any of the given values will be considered to belong to the segment.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIbmIbmAppConfigSegmentCreate(d *schema.ResourceData, meta interface{}) error {
+
+	guid := d.Get("guid").(string)
+	appconfigClient, err := getAppConfigClient(meta, guid)
+	if err != nil {
+		return err
+	}
+	options := &appconfigurationv1.CreateSegmentOptions{}
+	options.SetName(d.Get("name").(string))
+	options.SetSegmentID(d.Get("segment_id").(string))
+
+	if _, ok := d.GetOk("description"); ok {
+		options.SetDescription(d.Get("description").(string))
+	}
+	if _, ok := d.GetOk("tags"); ok {
+		options.SetTags(d.Get("tags").(string))
+	}
+
+	if _, ok := d.GetOk("rules"); ok {
+		var segmentRules []appconfigurationv1.Rule
+		for _, e := range d.Get("rules").([]interface{}) {
+			value := e.(map[string]interface{})
+			segmentRulesItem, err := resourceIbmAppConfigMapToSegmentRule(value)
+			if err != nil {
+				return err
+			}
+			segmentRules = append(segmentRules, segmentRulesItem)
+		}
+		options.SetRules(segmentRules)
+	}
+
+	segment, response, err := appconfigClient.CreateSegment(options)
+
+	if err != nil {
+		log.Printf("CreateSegment failed %s\n%s", err, response)
+		return err
+	}
+	d.SetId(fmt.Sprintf("%s/%s", guid, *segment.SegmentID))
+	return resourceIbmIbmAppConfigSegmentRead(d, meta)
+}
+
+func resourceIbmAppConfigMapToSegmentRule(segmentRuleMap map[string]interface{}) (appconfigurationv1.Rule, error) {
+	segmentRule := appconfigurationv1.Rule{}
+
+	segmentRule.AttributeName = core.StringPtr(segmentRuleMap["attribute_name"].(string))
+	segmentRule.Operator = core.StringPtr(segmentRuleMap["operator"].(string))
+	var values []string
+	for _, rulesItem := range segmentRuleMap["values"].([]interface{}) {
+		values = append(values, rulesItem.(string))
+	}
+	segmentRule.Values = values
+	return segmentRule, nil
+}
+
+func resourceIbmIbmAppConfigSegmentRead(d *schema.ResourceData, meta interface{}) error {
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return nil
+	}
+
+	appconfigClient, err := getAppConfigClient(meta, parts[0])
+	if err != nil {
+		return err
+	}
+
+	options := &appconfigurationv1.GetSegmentOptions{}
+	options.SetSegmentID(parts[1])
+
+	result, response, err := appconfigClient.GetSegment(options)
+	if err != nil {
+		return fmt.Errorf("[DEBUG] GetSegment failed %s\n%s", err, response)
```

Comment:
check for 404 and set id to "" for terraform to recreate when a segment is deleted outside terraform

Reply:
Addressed